science/shtools/Portfile

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+name                shtools
+revision            0
+
+categories          science math
+platforms           darwin
+license             BSD-3-Clause
+maintainers         {@MarkWieczorek gmail.com:mark.a.wieczorek}
+description         Spherical Harmonic Tools
+long_description    SHTOOLS is a library of Fortran 95 software that can be \
+                    used to perform spherical harmonic transforms, multitaper \
+                    spectral analyses, expansions of functions into Slepian \
+                    bases, and standard operations on global gravitational \
+                    and magnetic field data. Requires linking with FFTW-3 and \
+                    LAPACK compatible libraries.
+homepage            https://shtools.github.io/SHTOOLS/
+
+github.setup        SHTOOLS SHTOOLS 4.7.1 v
+github.tarball_from   archive
+checksums           sha256  176610743ddd76ea1fe7b45db3e8613171630b85c3759c9a4b31cde6b5d2ae3d \
+                    rmd160  58959a91c2889488844f6a7e4745090b0c906057 \
+                    size    35165653
+
+use_configure       no
+
+build.cmd           make
+build.target        fortran F95="${prefix}/bin/gfortran-mp-10"
+
+variant openmp description {Add OpenMP support} {
+    build.target-append     fortran-mp
+}
+
+test.run            yes
+test.cmd            make
+test.target         fortran-tests-no-timing
+test.args           PREFIX=${prefix} F95="${prefix}/bin/gfortran-mp-10" LAPACK="-framework Accelerate" BLAS=""
+
+destroot.cmd        make
+destroot.target     install
+destroot.args       PREFIX=${prefix}
+
+depends_build       port:gcc10 port:fftw-3
+depends_test        port:gcc10 port:fftw-3
+depends_lib         port:fftw-3
```

Comment:
I personally don't know the answer, but should this port also specify `libgcc10` as a library dependency?

Reply:
I have never had to specify `libgcc10` on linux or macOS before, and the builds below are passing. So my guess is No. The only reason we need `gcc10` at all is to install the `gfortran` compiler.

Reply:
> the builds below are passing. So my guess is No.

A successful CI build by no means guarantees that everything in a portfile is correct. But in this case it looks like the only binary file this port installs is the static library /opt/local/lib/libSHTOOLS.a. As such I agree that it doesn't have a library dependency libgcc10. And parts of libgcc10 that it needs will have been included within the static library.

Reply:
According to the instructions that the build prints, to use the library, one must link with `-lSHTOOLS -lfftw-3`. So the library dependency on fftw-3 is justified. And therefore, fftw-3 does not need to be specified as a build or test dependency; library dependencies are guaranteed to be present at build and test time.